### PR TITLE
Refactor enkf_main_write_run_path

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Continues the refactoring of `write_run_path` (https://github.com/equinor/ert/issues/3389) in preparation of https://github.com/equinor/ert/issues/3455

**Approach**
Localises parmeter and DATAKW writing in the enkf_main namespace temporarily. Refactors and documents current behavior.

Several future refactor work is left out to in order to have smaller incremental PRs:

* rename of ecl_write to something more descriptive (that name is used in several places)
* deep dive into called functions (primarily `forward_model_formatted_fprint`, `enkf_node_ecl_write` and `ert_template_instansiate`)
* Replace custom datatypes and utilities (mainly `util_make_path` and `run_context` arg list). 

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Review by oyvindeide — COMMENTED** [review]
Looks good, but looks like a lot of the code is shifted around? Would it be possible to trim down the diff? Would make it a bit easier to review.

File: libres/lib/enkf/enkf_main.cpp [hunk 1]
```diff
@@ -642,53 +634,92 @@ void enkf_state_ecl_write(const ensemble_config_type *ens_config,
...
+
+            ert_templates_instansiate(res_config_get_templates(res_config),
+                                      run_arg_get_runpath(run_arg),
+                                      run_arg_get_subst_list(run_arg));
+
+            ecl_write(ens_config,
+                      model_config_get_gen_kw_export_name(model_config),
+                      run_arg, run_arg_get_sim_fs(run_arg));
+
+            // Create the eclipse data file (if eclbase)
```
[the remark this answers is not in the document]
Comment (eivindjahren) [reply libres/lib/enkf/enkf_main.cpp]: Fixed :)

**Comment by eivindjahren** [comment]
> Looks good, but looks like a lot of the code is shifted around? Would it be possible to trim down the diff? Would make it a bit easier to review.

I split out one more commit in order to trim the diff as much as possible. Unfortunately there is some whitespace changes that git is not able to match up against.

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Pull request merged**